app/src/main/java/org/dhis2/Bindings/Bindings.java

```diff
@@ -648,4 +650,23 @@ public static void requestFocus(EditText editText, boolean focused) {
             editText.clearFocus();
         }
     }
+
+    @BindingAdapter("checkListener")
+    public static void checkListener(RadioGroup radioGroup, RadioButtonViewModel viewModel){
+        radioGroup.setOnCheckedChangeListener(null);
```

Comment:
do you need to disable the listener, then setup with the viewModel and then setup the listener?

Reply:
Yes.
You need to disable the listener every time a new viewModel is bound. If not, when setting the initial value the processor is triggered.